lunavl/sdk/estimators/face_estimators/head_pose.py

```diff
@@ -156,20 +156,45 @@ def estimate(self, landmarks68: Landmarks68) -> HeadPose:  # type: ignore
         return self.estimateBy68Landmarks(landmarks68)
 
     @CoreExceptionWrap(LunaVLError.EstimationHeadPoseError)
-    def estimateByBoundingBox(self, detection: BoundingBox, imageWithDetection: VLImage) -> HeadPose:
+    def estimateByBoundingBox(self, imageWithFaceDetection: ImageWithFaceDetection) -> HeadPose:
         """
         Estimate head pose by detection.
 
         Args:
-            detection: detection bounding box
-            imageWithDetection: image with the detection.
+            imageWithFaceDetection: image with face detection
         Returns:
             estimate head pose
         Raises:
             LunaSDKException: if estimation is failed
         """
-        error, headPoseEstimation = self._coreEstimator.estimate(imageWithDetection.coreImage, detection.coreEstimation)
+        error, headPoseEstimation = self._coreEstimator.estimate(
+            imageWithFaceDetection.image.coreImage, imageWithFaceDetection.boundingBox.coreEstimation
+        )
 
         if error.isError:
             raise LunaSDKException(LunaVLError.fromSDKError(error))
         return HeadPose(headPoseEstimation)
+
+    @CoreExceptionWrap(LunaVLError.EstimationHeadPoseError)
+    def estimateByBoundingBoxBatch(
+        self, imageWithFaceDetectionList: Union[List[ImageWithFaceDetection], List[FaceDetection]]
```

Comment:
FaceDetection is not an `image with face detection list`.  I think that need rename

Reply:
done

Reply:
i don't see.

Reply:
`List[ImageWithFaceDetection]` - `list of image with face detection`

`List[FaceDetection]` - `face detections`

or I don't understand something

Reply:
you named a variable `imageWithFaceDetectionList`. But List[FaceDetection] is not a image with face detection, it is detection with imae list. I suggest  renaming this variable so that both variant of using will be correct.

Reply:
renamed

Reply:
`faceDetectionOrImageWithFaceDetectionList` ))) mb neutral?  `batch` for example or something else?